lib/resourceful_endpoint.js

```diff
@@ -212,7 +212,67 @@ class ResourcefulEndpoint {
    * @returns {Promise} - A {@link ResourceCollection}
    */
   browse(criteria, options) {
-    return this.transport.get(this.endPointUrl(null, criteria), options).then(json => new ResourceCollection(json, criteria, this));
+    return this.transport
+      .get(this.endPointUrl(null, this.getRelatedThroughFields(criteria)), options)
+      .then(json => new ResourceCollection(json, criteria, this));
+  }
+
+  /**
+   * Private
+   * Look through the relationships of a criteria, and include fields from through
+   * relationships - this makes it so that the end user shouldn't know or care that
+   * through relationships exist.
+   *
+   * @param {(string|Query)} criteria - A query string to append to the request
+   *
+   * @returns {(Query)} criteria - A qquery with additional includes if needed
+
+   */
+  getRelatedThroughFields(criteria) {
```

Comment:
Does this belong here, or in the Query class?

Since it's modifying the criteria, should it be named `addRelatedThroughFields`?

Reply:
So it's been done here because we want to catch case where the query is passed as a simple string. I guess we could do something like:

```
const query = new Query(criteria);
query.addThroughRelationships();
```
?

That said the getThroughRelationship method appears to assume that criteria will *always* be a Query class despite allowing string values anyway? It sounds like somewhere we should convert criteria to a Query class?

```
browse(criteria, options) {
   let query = criteria;
    if (!criteria.hasOwnProperty('query')) {
      query = new Query(criteria);
    }
    query.addThroughRelationships();

    return this.transport
      .get(this.endPointUrl(null, query), options)
      .then(json => new ResourceCollection(json, criteria, this));
  }
```

Would something like this be better?

Reply:
Yeah, that makes it look a lot cleaner, and handles the String situation